### PR TITLE
[RN] Fix Samsung keyboard issue on the Chat screen

### PR DESCRIPTION
**Pull request by zbettenbuk**
This fixes an issue on Samsung phones with Samsung keyboard

### CODE REVIEW

File: react/features/chat/components/native/ChatInputBar.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    /**
+     * The value of the input field.
+     */
+    message: string
+};
+
+/**
+ * Implements the chat input bar with text field and action(s).
+ */
+export default class ChatInputBar extends Component<Props, State> {
```
Comment (saghul) [inline react/features/chat/components/native/ChatInputBar.js]: Let's call this `ChatInputField`, it's not a bar and there are references to "field" alesewhere here.

File: react/features/chat/components/native/ChatInputBar.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    _onSubmit: () => void;
+
+    /**
+     * Callback to handle the submit event of the text field.
+     *
+     * @returns {void}
+     */
+    _onSubmit() {
+        this.props.onSend((this.state.message || '').trim());
```
Comment (saghul) [inline react/features/chat/components/native/ChatInputBar.js]: I'd avoid calling onSend with an empty string.

File: react/features/chat/components/native/MessageContainer.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    /**
+     * Implements {@code Component#render}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        return (
+            <FlatList
+                data = { [ ...this.props.messages ].reverse() }
```
Comment (saghul) [inline react/features/chat/components/native/MessageContainer.js]: won't this create a copy of the array every time?

File: react/features/chat/components/native/MessageContainer.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    /**
+     * Renders a single chat message.
+     *
+     * @param {Object} message - The chat message to render.
+     * @returns {React$Element<*>}
+     */
+    _renderMessage({ item: message }) {
+        return (
+            <ChatMessage
+                key = { `key_${md5.hex(JSON.stringify(message))}` }
```
Comment (saghul) [inline react/features/chat/components/native/MessageContainer.js]: doesn't the message have any id of any kind?

File: react/features/chat/components/native/styles.js [hunk 5]
```diff
@@ -49,6 +50,22 @@ export default {
...
+    inputBar: {
+        borderTopColor: 'rgb(209, 219, 231)',
+        borderTopWidth: 1,
+        flexDirection: 'row',
+        paddingHorizontal: BoxModel.padding
+    },
+
+    inputField: {
+        flex: 1,
+        height: 48
```
Comment (saghul) [inline react/features/chat/components/native/styles.js]: is this related to MD? if so, a constant?

File: react/features/chat/components/native/MessageContainer.js (hunk 4 above)
Comment (zbettenbuk) [reply react/features/chat/components/native/MessageContainer.js]: Unfortunately not... we have a message.id, but that's the sending participant ID (bad design, in my opinion)

File: react/features/chat/components/native/styles.js (hunk 5 above)
Comment (zbettenbuk) [reply react/features/chat/components/native/styles.js]: No, this is from Zeplin. I don't thing "they" use MD at all

File: react/features/chat/components/native/Chat.js [hunk 6]
```diff
@@ -81,57 +73,8 @@ class Chat extends AbstractChat<Props> {
...
-                        originalText: message.message
-                    })
-                    : message.message,
-                user: {
-                    _id: message.id,
-                    name: message.displayName
-                }
-            }
-        );
+    _onSend(message) {
```
Comment (saghul) [inline react/features/chat/components/native/Chat.js]: why not remove this one and pass the prop directly to the ChatInputBar?

File: react/features/chat/components/native/ChatInputBar.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+    _onSubmit: () => void;
+
+    /**
+     * Callback to handle the submit event of the text field.
+     *
+     * @returns {void}
+     */
+    _onSubmit() {
+        const message = (this.state.message || '').trim();
```
Comment (saghul) [inline react/features/chat/components/native/ChatInputBar.js]: why the `||` AFAIS `this.state.message` cannot ever be null or undefined

File: react/features/chat/reducer.js [hunk 8]
```diff
@@ -22,14 +22,22 @@ ReducerRegistry.register('features/chat', (state = DEFAULT_STATE, action) => {
             timestamp: action.timestamp
         };
 
+        // React native, unlike web, needs a reverse sorted message list.
```
Comment (saghul) [inline react/features/chat/reducer.js]: I think we also need to change how lastReadMessage is computed, in the TOGGLE_CHAT action

File: react/features/chat/components/native/Chat.js (hunk 6 above)
Comment (zbettenbuk) [reply react/features/chat/components/native/Chat.js]: Hah, after all the refactors... Very true, thanks :)

File: react/features/chat/reducer.js (hunk 8 above)
Comment (zbettenbuk) [reply react/features/chat/reducer.js]: True, as well as the ```getUnreadCount``` function. Thanks!
